processor/metricsgenerationprocessor/utils.go

```diff
@@ -53,14 +53,23 @@ func generateMetrics(rm pmetric.ResourceMetrics, operand2 float64, rule internal
 			if metric.Name() == rule.metric1 {
 				newMetric := appendMetric(ilm, rule.name, rule.unit)
 				newMetric.SetEmptyGauge()
```

Comment:
Should we preserve the original type instead?

Reply:
I'm wondering how to properly set the type of the resulting metric based on input metric types. I think if both input metrics are the same type it's fair to give the resulting metric the same type, but I'm wondering more about how to surmise the resulting metric type when they're mixed. I've given examples below and will implement according to what I've shared in the summaries, but let me know if you think of a case I missed, or if anything is incorrect here.

## Non-commutative
### Examples
`sum` / `gauge`
```
experimental_metricsgeneration:
  rules:
    - name: system.filesystem.capacity
       unit: bytes
       type: calculate
       metric1: system.filesystem.usage # sum, bytes
       metric2: system.filesystem.utilization # gauge, percentage of bytes used
       operation: divide
```
The result would be the total bytes in a filesystem, which should be a sum.

`gauge` / `sum`
```
       metric1: read.rate # gauge, unit is bytes / second
       metric2: total.bytes.read # sum, unit is bytes
       operation: divide
```
This would result in percentage of total read bytes that were read in the last second (not necessarily a meaningful metric, but for the sake of example). This would be a gauge, I believe.

`sum` - `gauge`
```
       metric1: total.bytes.read # sum, unit is bytes
       metric2: read.rate # gauge, unit is bytes / second
       operation: subtract
```
Results in bytes read before the last second, a sum.

`gauge` - `sum`
```
       metric1: read.rate # gauge, unit is bytes / second
       metric2: file.size # sum, unit is bytes
       operation: subtract
```
Result would be gauge, how many bytes of the file weren't read in a second (once again, not necessarily important or meaningful, but valid).

### Summary
For non-commutative examples, the resulting metric type could simply be the type of `metric1`.

## Commutative
### Examples

`gauge` + `sum`
```
       metric1: read.rate # gauge, unit is bytes / second
       metric2: total.bytes.read # sum, unit is bytes
       operation: add
```
Result would be an estimation of how many total bytes would be read after another second, a sum. The resulting type should be a sum, regardless of order.

`gauge` * `sum`
```
       metric1: read.rate # gauge, unit is bytes / second
       metric2: device.count # sum, unit is devices
       operation: multiply
```
Result would be an estimation of how many bytes could be read per second across the whole system, a gauge.

### Summary

From these examples I think it depends on calculation type. For `add`, if either metric is a `sum`, the result should be a `sum`. For `multiply`, if either metric is a `gauge`, the result should be a gauge.

Reply:
I'm not sure we need to support cross-type calculation, but for simplicity I would just inherit the type from the first operand. We can clearly document that. It should suit most of the use cases. WDYT?

Reply:
I think that's a fair compromise. Documenting something slightly unexpected is probably a better UX than the alternative here. 👍